glean-core/src/ping/mod.rs

```diff
@@ -93,8 +93,10 @@ impl PingMaker {
     }
 
     pub fn collect_string(&self, storage: &Database, storage_name: &str) -> Option<String> {
+        // Use ::to_string, rather than ::to_string_pretty since the ping
+        // uploader relies on the JSON content being on a single line.
```

Comment:
Ah, good to know. Well, `pretty` was so far mostly so that we see it nicely in the log output anyway.

Reply:
That's a good point.  We may want to use pretty for logging only...

Reply:
@badboy: I've updated this so:
1) Serialization of the JSON happens in `storePing` so there's a single place that is responsible for *exactly* how the ping is serialized.
2) When logging the ping, we still use `pretty`.  (Yes, that means we serialize it twice, but logging pings is a QA-only sort of thing -- that's better than glean-ac where we serialize->deserialize->pretty, I guess)

This meant I made a bigger change and added a new Error type and stuff, so probably worth another quick review of the last commit here.

Reply:
Is `collect_string` anywhere still used? If not, should we just remove it?